### DISCUSSION TITLE
How to new field to specific connection nodes

### DISCUSSION THREAD

**Discussion by PeterSchuhmannNetflow** [wp-graphql/wp-graphql#2884] [General]
Hello everyone and thank you for your great work with wp-graphql. Everything is well documented and works like a charme. But I found one case where I got a little confused or couldn't figure out a solution.

Problem: I would like to register custom fields to a Post that are only available for a specific query. Lets say you have a query like this:
```
{
  postsSearch(
    where: {search: "wordpress", offsetPagination: {offset: 0, size: 10}}
  ) {
    nodes {
      __typename
      title
      slug
    }
  }
}
``` 

nodes contains the regular Post object and all fields related to it. I would like to add more fields to `nodes` that are only available when using `postsSearch` like this:

```
{
  postsSearch( 
    where: {search: "wordpress", offsetPagination: {offset: 0, size: 10}}
  ) {
    nodes {
      __typename
      title
      slug
      additionalField1
      additionalField2
    }
  }
}
``` 

these new fields should not be available when using a regular post query. 

I am not sure where i went wrong maybe you guys could help.

**Comment by justlevine** [comment]
Hey @PeterSchuhmannNetflow 

In GraphQL, types and schemas are supposed to be _static_, meaning their shape should not change conditionally, so it makes sense you're running into trouble.

Instead, you can
- use directives in a query to skip or include the resolution of fields
-  put the fields on an Edge (if they exist to describe the connection to the Type, and aren't a property _of_ the type), 
- or add them to the Type's model and have them return null under some circumstances.
